File: AiSD5_C2/exercise_1/Pascal/main.pas
program RownanieKwadratowe;

procedure CzytajDane(var a, b, c: real);
begin
  writeln('Czytaj: a, b, c');
  write('a: ');
  readln(a);
  write('b: ');
  readln(b);
  write('c: ');
  readln(c);
end;

function ObliczRownanieDelta(a, b, c: real; var x1, x2: real): boolean;
var
  Delta: real;
begin
  Delta := b * b - 4 * a * c;
  if Delta > 0 then
  begin
    x1 := (-b - sqrt(Delta)) / (2 * a);
    x2 := (-b + sqrt(Delta)) / (2 * a);
    ObliczRownanieDelta := TRUE;
  end
  else if Delta = 0 then
  begin
    x1 := -b / (2 * a);
    x2 := x1;
    ObliczRownanieDelta := TRUE;
  end
  else
    ObliczRownanieDelta := FALSE;
end;

procedure PiszWyniki(x1, x2: real; CzyPisac: boolean);
begin
  writeln('Wyniki obliczen');
  if CzyPisac then
  begin
    if x1 <> x2 then
    begin
      writeln('Dwa pierwiastki: x1 = ', x1:6:2, ', x2 = ', x2:6:2);
    end
    else
    begin
      writeln('Jeden pierwiastek: x = ', x1:6:2);
    end;
  end
  else
    writeln('Brak rozwiazan.');
end;

procedure Rownanie;
var
  a, b, c, x, x1, x2: real;
  CzyJestRozwiazanie: boolean;
begin
  CzytajDane(a, b, c);
  if a = 0 then
  begin
    if b <> 0 then
    begin
      x := -c / b;
      writeln('To jest rownanie liniowe, x = ', x:6:2);
    end
    else
      writeln('To nie jest rownanie.');
  end
  else
  begin
    CzyJestRozwiazanie := ObliczRownanieDelta(a, b, c, x1, x2);
    PiszWyniki(x1, x2, CzyJestRozwiazanie);
  end;
end;

begin
  writeln('Rozwiazanie rownania kwadratowego');
  Rownanie;
end.

File: AiSD5_C2/exercise_1/Python/main.py
import math

def czytaj_dane():
    print('Czytaj: a, b, c')
    a = float(input('a: '))
    b = float(input('b: '))
    c = float(input('c: '))
    return a, b, c

def oblicz_rownanie_delta(a, b, c):
    delta = b * b - 4 * a * c
    if delta > 0:
        x1 = (-b - math.sqrt(delta)) / (2 * a)
        x2 = (-b + math.sqrt(delta)) / (2 * a)
        return True, x1, x2
    elif delta == 0:
        x1 = -b / (2 * a)
        return True, x1, None
    else:
        return False, None, None

def pisz_wyniki(x1, x2, czy_pisac):
    print('Wyniki obliczen')
    if czy_pisac & (x2 != None):
        print(f'Dwa pierwiastki: x1 = {x1:.2f}, x2 = {x2:.2f}')
    elif czy_pisac & (x2 == None):
        print(f'Jeden pierwiastek: x = {x1:.2f}')
    else:
        print('Brak rozwiazan.')

def rownanie():
    a, b, c = czytaj_dane()
    if a == 0:
        if b != 0:
            x = -c / b
            print(f'To jest rownanie liniowe, x = {x:.2f}')
        else:
            print('To nie jest rownanie.')
    else:
        czy_jest_rozwiazanie, x1, x2 = oblicz_rownanie_delta(a, b, c)
        pisz_wyniki(x1, x2, czy_jest_rozwiazanie)

if __name__ == "__main__":
    print('Rozwiazanie rownania kwadratowego')
    rownanie()